google-cloud-storage/src/main/java/com/google/cloud/storage/GapicUnbufferedReadableByteChannel.java

```diff
@@ -234,35 +234,42 @@ public void close() throws IOException {
   }
 
   private void drainQueue() throws IOException {
-    IOException ioException = null;
-    while (queue.nonEmpty()) {
-      try {
-        java.lang.Object queueValue = queue.poll();
-        if (queueValue instanceof ReadObjectResponse) {
-          ReadObjectResponse resp = (ReadObjectResponse) queueValue;
-          ResponseContentLifecycleHandle handle = rclm.get(resp);
-          handle.close();
-        } else if (queueValue == EOF_MARKER || queueValue instanceof Throwable) {
-          break;
-        }
-      } catch (IOException e) {
-        if (ioException == null) {
-          ioException = e;
-        } else if (ioException != e) {
-          ioException.addSuppressed(e);
+    boolean shouldInterupt = false;
+    try {
+      IOException ioException = null;
+      while (queue.nonEmpty()) {
+        try {
+          java.lang.Object queueValue = queue.poll();
+          if (queueValue instanceof ReadObjectResponse) {
+            ReadObjectResponse resp = (ReadObjectResponse) queueValue;
+            ResponseContentLifecycleHandle handle = rclm.get(resp);
+            handle.close();
+          } else if (queueValue == EOF_MARKER || queueValue instanceof Throwable) {
+            break;
+          }
+        } catch (IOException e) {
+          if (ioException == null) {
+            ioException = e;
+          } else if (ioException != e) {
+            ioException.addSuppressed(e);
+          }
+        } catch (InterruptedException e) {
+          shouldInterupt = true;
+          if (ioException == null) {
+            ioException = new InterruptedIOException();
+          } else {
+            ioException.addSuppressed(e);
+          }
         }
-      } catch (InterruptedException e) {
+      }
+      if (ioException != null) {
```

Comment:
why did this get moved out of the finally block?

Reply:
This conditional was already a sibling to the while loop.

The try-finally was added in this change. We specially want to make sure that the thread interrupted state is propagated as the final step of our method.

Reply:
ah sorry i missed that. 👍